Use a confluence module to update status pages

romanofski opened the pull request:
This adds a module which allows the user to forward the generated confluence
template to Confluence in order to replace the content up there. The flow at the
moment is:

* log in (using Kerberos)
* convert the wiki markup from finishline to Confluence storage format
* post the converted content to Confluence

Fixes: https://github.com/ralphbean/finishline/issues/38

romanofski commented:
Perhaps not just merge away, but try it first ...

accorvin commented:
This code looks fine to me. I'm not really sure how to test is though. Can you post some instructions for how to do so?

romanofski commented:
My workflow is:
* finishline renders the confluence template
* confluence.py takes the rendered output and replaces the page given by it's ID

Note I only needed Kerberos to log in under my name so I haven't added any additional mechanisms.

I've extended the example confluence template to be like:
```
{excerpt:hidden=false|atlassian-macro-output-type=BLOCK}

h2. [reporting:Beaker Project Status]

{info:title=Project Info}
Last Updated: {{today}}
Overall Status:
Business Value: Hardware testing and lab management
Product Owner:  [~rjoost]
Project Landing Page: [HTD:Beaker]
{info}

h3. In Progress/Completed in last month

||Status||Owner||Task||Original Date||Actual Date||Notes||Progress||
{% for objective in objectives | sort %}|| || || {{ objective }} || || || || ||
{% for key, epic in epics.items() if epic and key in objectives[objective] %}|{status:colour={{epic.mvp_status['statusCategory']['colorName']}}|title={{epic.mvp_status['name']}}}| [~{{epic.owner['name']}}] | [{{epic.raw['fields']['summary'].split('\n')[0].split('KR:')[-1].strip() }}|https://projects.engineering.redhat.com/browse/{{key}}|{{key}}] | {{epic.target_date}} | {{epic.target_date}} | {{epic.status_update.cleaned}} | {{epic.percent_complete}}% |
{% endfor %}{% endfor %}

h3. Next Quarters Priorities
{jira:key=BKR-3843}
{jira:key=BKR-2757}
{excerpt}
```
since that goes to a dedicated page aggregated by a "dashboard" page.

I then checked the page ID for the page I want to always overwrite, since the entire page contents are replaced. You can find this when you click on the *Page Information* link. The page ID is then found in the URL. Perhaps there is a better way, but that was what worked for me.

So all of that is then part of my Makefile target:

```Makefile
confluence: install                                                                                                                                                                                                                          
  venv/bin/python finishline \
    --server https://projects.engineering.redhat.com \
    --project BKR \
    --title "Beaker, OKRs" \
    --hide-epics=BKR-3934 \
    --include-epics=BKR-3843,BKR-2891 \
    --mvp-status-field=status \
    --auth-path=/step-auth-gss \
    --since "$(shell LC_ALL='en_US.UTF-8' date -d '1 month ago' '+%Y-%m-%d')" \
    --template romanofski/confluence.something > confluence.something
  venv/bin/python confluence.py \
    --server https://docs.engineering.redhat.com \
    --pageid 38176033 \
    --auth-path=/step-auth-gss \
    confluence.something
  echo " -- done -- "
```

Let me know if that helps. Happy to explain more.
